usermods/wireguard/wireguard.h

```diff
@@ -0,0 +1,102 @@
+#pragma once
+
+#include <WireGuard-ESP32.h>
+
+#include "wled.h"
+
+class WireguardUsermod : public Usermod {
+   public:
+    void setup() { configTzTime(posix_tz.c_str(), ntpServerName); }
+
+    void connected() {
+        if (wg.is_initialized()) {
+            wg.end();
+        }
+    }
+
+    void loop() {
+        struct tm timeinfo;
+        if (!is_enabled || !getLocalTime(&timeinfo, 0)) {
+            return;
+        }
+
+        if (wg.is_initialized() || !WLED_CONNECTED) {
+            return;
+        }
+
+        if (preshared_key.length() < 5) {
+            wg.begin(local_ip, private_key.c_str(), endpoint_address.c_str(), public_key.c_str(), endpoint_port, NULL);
+        } else {
+            wg.begin(local_ip, private_key.c_str(), endpoint_address.c_str(), public_key.c_str(), endpoint_port, preshared_key.c_str());
+        }
+    }
+
+    void addToJsonInfo(JsonObject& root) {
+        JsonObject user = root["u"];
+        if (user.isNull()) user = root.createNestedObject("u");
+
+        JsonArray infoArr = user.createNestedArray(F("WireGuard VPN"));
+        String uiDomString;
+        
+        struct tm timeinfo;
+        if (!getLocalTime(&timeinfo, 0)) {
+            uiDomString = "Time out of sync!";
+        } else {
+            if (wg.is_initialized()) {
+                uiDomString = "netif up!";
+            } else {
+                uiDomString = "netif down :(";
+            }
+        }
+        if(is_enabled) infoArr.add(uiDomString);
+    }
+
+    void addToConfig(JsonObject& root) {
+        JsonObject top = root.createNestedObject(F("WireGuard VPN"));
+        top[F("Endpoint Address")] = endpoint_address;
```

Comment:
Please avoid key names with spaces and make them short if possible.
You can use utility function `addInfo()` to make UI better.